xdsl/dialects/builtin.py

```diff
@@ -395,6 +396,23 @@ def verify_value(self, value: int):
                 f"values in the range [{min_value}, {max_value})"
             )
 
+    def normalize_value(self, value: int) -> int:
+        """
+        Normalize signless values within the expected range to the signed range for the
+        same bitwidth.
+        Values outside the valid range are left as-is.
+        """
+        if self.signedness.data != Signedness.SIGNLESS:
+            return value
+
+        signed_ub = signed_upper_bound(self.bitwidth)
+        unsigned_ub = signed_ub << 1
+
+        if signed_ub <= value < unsigned_ub:
```

Comment:
```suggestion
        if signed_ub < value < unsigned_ub:
```
signed_ub is exclusive right?

Reply:
it's exclusive on the other side, meaning that a value that is ub is not within the accepted range, and should be changed

Reply:
Of course